xline/src/server/kv_server.rs

```diff
@@ -42,30 +47,39 @@ where
     id_barrier: Arc<IdBarrier>,
     /// Range request retry timeout
     range_retry_timeout: Duration,
+    /// Compact timeout
+    compact_timeout: Duration,
     /// Consensus client
     client: Arc<Client<Command>>,
+    /// Compact events
+    compact_events: Arc<RwLock<HashMap<ProposeId, Arc<Event>>>>,
 }
 
 impl<S> KvServer<S>
 where
     S: StorageApi,
 {
     /// New `KvServer`
+    #[allow(clippy::too_many_arguments)]
     pub(crate) fn new(
```

Comment:
Shall we use the builder pattern to refactor this `new` method?

Reply:
This method will only be called once, and the other server implementations are the same, which I think is better if the interface is consistent